code/__HELPERS/medals.dm

```diff
@@ -0,0 +1,119 @@
+//It also covers stuff for keeping track of certain statistics.
+
+#define ANNOUNCE_TO_NOBODY "nobody" //Silent, doesn't tell anyone
+#define ANNOUNCE_TO_PLAYER "player" //Tells the person who earned the achievement
+#define ANNOUNCE_TO_ADMINS "admins" //Tells the unlocker and any online admins
+#define ANNOUNCE_TO_EVERYONE "everyone" //Tells everyone
+
+/proc/CheckMedal(medal,client/player) //Checks if a player has a medal, returning 1 on a success and 0 on a failure
+
+	if(!player || !medal)
+		return
+	if(global.medal_hub && global.medal_pass && global.medals_enabled)
+
+		var/result = world.GetMedal(medal, player, global.medal_hub, global.medal_pass)
+
+		if(isnull(result))
+			global.medals_enabled = FALSE
+			log_game("MEDAL ERROR: Could not contact hub to get medal:[medal] player:[player.ckey]")
+			message_admins("Error! Failed to contact hub to get [medal] medal for [player.ckey]!")
+			return 0
+		else if (result)
+			return TRUE
+		return FALSE
+
+
+/proc/UnlockMedal(medal, client/player, announce = ANNOUNCE_TO_PLAYER) //Adds a medal to the player
+	if(!player || !medal)
+		return
+	if(global.medal_hub && global.medal_pass && global.medals_enabled)
+		spawn()
+			var/result = world.SetMedal(medal, player, global.medal_hub, global.medal_pass)
+			if(isnull(result))
+				global.medals_enabled = FALSE
+				log_game("MEDAL ERROR: Could not contact hub to award medal:[medal] player:[player.ckey]")
+				message_admins("Error! Failed to contact hub to award [medal] medal to [player.ckey]!")
+			else
+				log_game("Player [player.ckey] unlocked medal [medal].")
+				switch(announce)
+					if(ANNOUNCE_TO_PLAYER)
+						player << "<span class='noticealien'>Achievement unlocked:</span> <span class='alertalien'>[medal]</span><span class='noticealien'>!</span>"
+					if(ANNOUNCE_TO_ADMINS)
+						player << "<span class='noticealien'>Achievement unlocked:</span> <span class='alertalien'>[medal]</span><span class='noticealien'>!</span>"
+						message_admins("Player [player.ckey] unlocked medal [medal]!")
+					if(ANNOUNCE_TO_EVERYONE)
+						player << "<span class='noticealien'>[player.ckey] has earned the</span> <span class='alertalien'>[medal]</span> <span class='noticealien'>achievement!</span>"
+
+
+/proc/LockMedal(medal, client/player, announce = ANNOUNCE_TO_ADMINS) //Removes a medal from the player
+	if(!player || !medal)
+		return
+	if(global.medal_hub && global.medal_pass && global.medals_enabled)
+		var/result = world.ClearMedal(medal, player, global.medal_hub, global.medal_pass)
+		if(isnull(result))
+			global.medals_enabled = FALSE
+			log_game("MEDAL ERROR: Could not contact hub to clear medal:[medal] player:[player.ckey]")
+			message_admins("Error! Failed to contact hub to clear [medal] medal for [player.ckey]!")
+		else if(result)
+			message_admins("Medal: [medal] removed for [player.ckey]")
+		else
+			log_game("Player [player.ckey] unlocked medal [medal].")
```

Comment:
That's a log for UNlocking a medal in the LOCK proc. Where the rest of the stuff says lock as well.